sklearn/tree/_criterion.pyx

```diff
@@ -261,7 +259,7 @@ cdef class ClassificationCriterion(Criterion):
         y : array-like, dtype=DOUBLE_t
             The target stored as a buffer for memory efficiency
         sample_weight : array-like, dtype=DOUBLE_t
```

Comment:
It might be a bit worth updating the docstring mentioning that we use memoryviews for `y` and `sample_weight`.

Reply:
or `ndarray` since we use it elsewhere

Reply:
Done.